cpg_workflows/stages/gatk_sv/gatk_sv_multisample_2.py

```diff
@@ -45,12 +42,23 @@
 # create the file path outside Stages, so that
 # we can pass this to the metadata update function
 RUN_DATETIME = datetime.now().strftime('%Y-%m-%d')
-EXCLUSION_FILE = join(
-    get_config()['storage']['default']['default'],
-    'gatk_sv',
-    RUN_DATETIME,
-    'combined_exclusion_list.txt',
-)
+
+
+@lru_cache(maxsize=1)
+def get_exclusion_filename() -> str:
```

Comment:
This feels heavy. Will this be an unusual usage or are we going to end up with needing to define getters everywhere? If so could we bake in some tooling to get_config somehow to handle this situation?

Reply:
The way that I'd written this initially (plus the mito changes) was the only instance of generating a variable at the top level of a file. A getter isn't really necessary, it's just building a file path so it could be done manually instead of calling this method.

John was having issues running local tests, given that this makes some assumptions about the config content, and this ran when the module was imported in main.py. This should have been fine with a config stub, but it's still rough design putting this variable assignment where it was going to be generated immediately even if it's never used.

Reply:
Most of these things just build up these filenames inside the methods, so the `get_config()` call doesn't happen until the particular pipeline is run. But these ones have it as a variable at the top-level, so the calls happen at import time in _main.py_ — which explodes for me and others on local tests. This is what Cas and I talked about in _mito.py_ many months ago, but you and I then forgot about it again :smile:

I guess these two files have them as top-level variables because the (sets of) filenames are wildly used so it's easier than repeating the definitions…